### PR TITLE
[TH2-2975] Escape restricted characters in Book's keyspace name

### PR DESCRIPTION
**Pull request by std000**

### CODE REVIEW

File: cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java [hunk 1]
```diff
@@ -888,4 +897,13 @@ private boolean checkFilter(TestEventFilter filter) throws CradleStorageExceptio
 		
 		return true;
 	}
+
+	protected void checkBookToAdd(BookToAdd book) throws CradleStorageException {
+		if (book.getName() == null || !BOOK_PAGE_NAME_PATTERN.matcher(book.getName()).matches()) {
+			throw new CradleStorageException("Invalid book name. Alphanumeric characters and underscore are allowed.");
+		}
+		if (book.getFirstPageName() == null || !BOOK_PAGE_NAME_PATTERN.matcher(book.getFirstPageName()).matches()) {
```
Comment (MrJonniK) [inline cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java:905]: why is the page name so strictly limited?

File: cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java [hunk 2]
```diff
@@ -817,6 +822,10 @@ public final void updateEventStatus(StoredTestEvent event, boolean success) thro
 		List<PageInfo> result = new ArrayList<>(pages.size());
 		for (PageToAdd page : pages)
 		{
+			if (page.getName() == null || !BOOK_PAGE_NAME_PATTERN.matcher(page.getName()).matches()) {
+				throw new CradleStorageException("Invalid page name. Alphanumeric characters and underscore are allowed.");
```
Comment (Vladimir-Panarin-Exactpro) [inline cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java:826]: Show page name here.

File: cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java [hunk 3]
```diff
@@ -817,6 +822,10 @@ public final void updateEventStatus(StoredTestEvent event, boolean success) thro
 		List<PageInfo> result = new ArrayList<>(pages.size());
 		for (PageToAdd page : pages)
 		{
+			if (page.getName() == null || !BOOK_PAGE_NAME_PATTERN.matcher(page.getName()).matches()) {
```
Comment (Vladimir-Panarin-Exactpro) [inline cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java:825]: Move this to separate method `validatePageName(String name)`?

File: cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java (hunk 1 above)
Comment (Vladimir-Panarin-Exactpro) [inline cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java:905]: Use new `validatePageName(String name)` method here.

**Review by Vladimir-Panarin-Exactpro — CHANGES_REQUESTED** [review]

File: cradle-core/src/test/java/com/exactpro/cradle/CradleStorageTest.java [hunk 4]
```diff
@@ -83,4 +83,89 @@ public void storeEventUnknownBook(TestEventSingleToStoreBuilder builder, String
 			TestUtils.handleException(e, errorMessage);
 		}
 	}
+
+	@Test(expectedExceptions = {CradleStorageException.class}, expectedExceptionsMessageRegExp = "Invalid book name.*")
```
Comment (Vladimir-Panarin-Exactpro) [inline cradle-core/src/test/java/com/exactpro/cradle/CradleStorageTest.java:87]: I propose to move all names-related tests to separate class.

**Review by Vladimir-Panarin-Exactpro — APPROVED** [review]
Despite of proposal to move name-related to tests to separate class I don't have objections.

**Pull request merged**